Lib/asyncore.py

```diff
@@ -181,11 +178,12 @@ def poll2(timeout=0.0, map=None):
                 pollster.register(fd, flags)
 
         r = pollster.poll(timeout)
-        for fd, flags in r:
-            obj = map.get(fd)
-            if obj is None:
-                continue
-            readwrite(obj, flags)
+        ready = [(map[fd], flags) for fd, flags in r]
+
+        for obj, flags in ready:
+            if not obj.closing:
```

Comment:
Would you mind to first address the first bug in a first PR, map modified in this loop, and *then* work on a new PR (once the first PR is merged) to add closing?

Again, I prefer very small changes.